projects/RabbitMQ.Client/client/framing/BasicProperties.cs

```diff
@@ -56,85 +61,198 @@ internal sealed class BasicProperties : RabbitMQ.Client.Impl.BasicProperties
 
         public override string ContentType
         {
-            get => _contentType;
+            get
+            {
+                if (_contentType == default && IsContentTypePresent())
+                {
+                    WireFormatting.ReadShortstr(_readOnlyMemory.Span.Slice(ContentTypeOffset), out _contentType);
+                }
+
+                return _contentType;
+            }
             set => _contentType = value;
         }
 
         public override string ContentEncoding
         {
-            get => _contentEncoding;
+            get
+            {
+                if (_contentEncoding == default && IsContentEncodingPresent())
+                {
+                    WireFormatting.ReadShortstr(_readOnlyMemory.Span.Slice(ContentEncodingOffset), out _contentEncoding);
+                }
+
+                return _contentEncoding;
+            }
             set => _contentEncoding = value;
         }
 
         public override IDictionary<string, object> Headers
         {
-            get => _headers;
+            get
+            {
+                if (_headers == default && IsHeadersPresent())
+                {
+                    WireFormatting.ReadDictionary(_readOnlyMemory.Span.Slice(HeadersOffset), out Dictionary<string, object> tempDirectory);
+                    _headers = tempDirectory;
```

Comment:
Temp Directory?

Reply:
IDictionary<,> vs Dictionary<,> :( Unfortunately we are exposing IDictionary<,> interfaces in a few places and it isn't co/contravariant :(

Reply:
tempDirectory, can we rename to tempDictionary.

Reply:
@stebet I think folks are simply pointing out that there is a local variable naming typo :)